src/test/javascript/spec/component/complaints/complaint.service.spec.ts

```diff
@@ -0,0 +1,329 @@
+import { TestBed } from '@angular/core/testing';
+import { ComplaintService } from 'app/complaints/complaint.service';
+import { Complaint, ComplaintType } from 'app/entities/complaint.model';
+import { TextExercise } from 'app/entities/text-exercise.model';
+import { ComplaintResponse } from 'app/entities/complaint-response.model';
+import { User } from 'app/core/user/user.model';
+import { AccountService } from 'app/core/auth/account.service';
+import { MockAccountService } from '../../helpers/mocks/service/mock-account.service';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import dayjs from 'dayjs/esm';
+import { Result } from 'app/entities/result.model';
+
+describe('ComplaintService', () => {
+    let complaintService: ComplaintService;
+    let accountService: AccountService;
+    let httpMock: HttpTestingController;
+
+    const dayjsTime1 = dayjs().utc().year(2022).month(3).date(14).hour(10).minute(35).second(12).millisecond(332);
+    const stringTime1 = '2022-04-14T10:35:12.332Z';
+    const dayjsTime2 = dayjs().utc().year(2022).month(4).date(12).hour(18).minute(12).second(11).millisecond(140);
+    const stringTime2 = '2022-05-12T18:12:11.140Z';
+
+    const clientComplaint1 = new Complaint();
+    clientComplaint1.id = 42;
+    clientComplaint1.complaintType = ComplaintType.MORE_FEEDBACK;
+    clientComplaint1.result = new Result();
+    clientComplaint1.submittedTime = dayjsTime1;
+    clientComplaint1.complaintText = 'Test text';
+
+    const serverComplaint1 = { ...clientComplaint1, submittedTime: stringTime1 };
+
+    const clientComplaint2 = new Complaint();
+    clientComplaint2.id = 42;
+    clientComplaint2.complaintType = ComplaintType.MORE_FEEDBACK;
+    clientComplaint2.result = new Result();
+    clientComplaint2.submittedTime = dayjsTime2;
+    clientComplaint2.complaintText = 'Another test text';
+
+    const serverComplaint2 = { ...clientComplaint2, submittedTime: stringTime2 };
+
+    beforeEach(() => {
+        return TestBed.configureTestingModule({
```

Comment:
```suggestion
        TestBed.configureTestingModule({
```
As in the other file.